### ISSUE TITLE
BUG: 0th power of sparse array/matrix always returns the identity of float64

### ISSUE THREAD

**Issue by takagi** [scipy/scipy#15224]
### Describe your issue.

The power of sparse array/matrix is expected to return one with the same dtype. However, in SciPy 1.8.0rc1, its 0th power always returns the identity of float64. I guess it was unintentionally missing in https://github.com/scipy/scipy/commit/c3e4ec597d857bcd5e37499622e2a2adcf697713.

### Reproducing Code Example

```python
>>> import numpy as np
>>> import scipy.sparse as sp
>>> a = sp.coo_array(np.ones((2, 2), dtype='D'))
>>> a.dtype
dtype('complex128')
>>> (a**0).dtype
dtype('float64')
```

### Error message

```shell
N/A
```

### SciPy/NumPy/Python version information

1.8.0rc1 1.22.0rc1 sys.version_info(major=3, minor=9, micro=7, releaselevel='final', serial=0)

**Comment by takagi** [comment]
xref: https://github.com/cupy/cupy/issues/6233

**Comment by tylerjereddy** [comment]
@stefanv @WarrenWeckesser should this be on the radar for `1.8.0` as well?

**Comment by WarrenWeckesser** [comment]
@tylerjereddy, I think so.  Ideally any PRs that fix problems discovered in the release candidate that were missed in the review of https://github.com/scipy/scipy/pull/14822 should be targeted for 1.8.0.  I'm not singling out  https://github.com/scipy/scipy/pull/14822 for special treatment; this  applies to any PRs that fix problems with new 1.8 features that are discovered in the release candidate, such as https://github.com/scipy/scipy/pull/15220, which fixes a small issue that is the result of the big namespace clean up. (That particular fix, however, is definitely not a high priority, so bumping it to 1.8.1 would be fine.)

**Comment by tylerjereddy** [comment]
Right, I just meant to assess priority.

Regardless, this seems to be fixed by the now-merged gh-15225
